examples/repl.rs

```diff
@@ -431,6 +431,144 @@ mod example {
+            )
+            .collect(),
+        };
+        Ok(call_stack)
+    }
+
+    /// Get the call_stack from the context and then tries to display the
+    /// source for the top call in the stack. Becasue the first frame is usually
+    /// sse2.rs, we just display the file and line but not the source and we skip
+    /// over to the next frame. For the next frame, we will display the source code.
+    /// An example view is shown below:
+    ///
+    /// It marks the line with the berakpoint with a '>' character and shows some lines
+    /// of context above and below it.
+    ///
+    /// ```plain
+    /// 0000555555559295 core::core_arch::x86::sse2::_mm_pause /../rustup/toolchains/1.45.2-x86_64-unknown-linux-gnu/../stdarch/crates/core_arch/src/x86/sse2.rs:25
+    /// /workspaces/headcrab/tests/testees/hello.rs:7:14
+    ///    4 #[inline(never)]
+    ///    5 fn breakpoint() {
+    ///    6     // This will be patched by the debugger to be a breakpoint
+    /// >  7     unsafe { core::arch::x86_64::_mm_pause(); }
+    ///    8 }
+    ///    9
+    ///   10 #[inline(never)]
+    /// ```
+    fn print_source_for_top_of_stack_symbol(
+        context: &mut Context,
+        context_lines: usize,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        let call_stack = get_call_stack(context, BtType::FramePtr)?;
+        let top_of_stack = call_stack[0];
+        context
+            .debuginfo()
+            .with_addr_frames(top_of_stack, |_addr, mut frames| {
+                while let Some(frame) = frames.next()? {
+                    let name = frame
+                        .function
+                        .as_ref()
+                        .map(|f| Ok(f.demangle()?.into_owned()))
+                        .transpose()
+                        .map_err(|err: gimli::Error| err)?
+                        .unwrap_or_else(|| "<unknown>".to_string());
+
+                    // TODO: currently we are skipping over mm_pause but once we have real
+                    // breakpoint support and _patch_breakpoint_function gets removed, this
+                    // must also be removed.
+                    if !name.contains("_mm_pause") {
+                        let (file, line, column) = frame
+                            .location
+                            .as_ref()
+                            .map(|loc| {
+                                (
+                                    loc.file.unwrap_or("<unknown file>"),
+                                    loc.line.unwrap_or(0),
+                                    loc.column.unwrap_or(0),
+                                )
+                            })
+                            .unwrap_or_default();
+
+                        print_file_lines_with_context(
+                            file,
+                            line as usize,
+                            column,
+                            context_lines as usize,
+                        )?;
```

Comment:
Please ads a return here to avoid printing multiple source lines in case of inlining.

Reply:
I was thinking about it.Thanks for pointing this out.